main.py

```diff
@@ -73,72 +79,221 @@ def __init__(self, json):
 
 
 
-# Check if all block heighs are in sync
+# Check if all block heights are in sync
 def dump_consensus(i):
     #create class
     class Dump():
         def __init__(self, json):
             self.dump=json["result"]["peer_round_states"]
 
     url_data = json.load(urllib2.urlopen("http://"+ i + "/dump_consensus_state"))
+
+    #get peer round states to check what validators think the block heigh should be
     state = Dump (url_data)
     for k in node_ip:
-        # print state.dump[k]['Height']
-        if state.dump[k]['Height'] != block_height+1:
-            print colored("Block height different - " + k + ' ' + "(height " + str(state.dump[k]['Height'])+ ")", 'red')
-        else:
+        try:
+            # compare peer round states to actual block height for consensus
+            if state.dump[k]['Height'] != block_height+1:
+                print colored("Block height different - " + k + ' ' + "(height " + str(state.dump[k]['Height'])+ ")", 'red')
+            else:
+                pass
+        except:
+            pass
+        try:
+            if state.dump["891023d33e161bafff356b74ea44730d295342b9"]['Height'] != block_height+1:
+                print colored("Block height different - " + ' ' + "(height " + str(state.dump["891023d33e161bafff356b74ea44730d295342b9"]['Height'])+ ")", 'red')
+            else:
+                pass
+        except:
             pass
 
+
+
+# saved in db
+last_run = 18089
+
 # Scan all blocks
 def scan(i):
-    #create class
-    class Block():
+    #create classes
+    class BlockHeight():
         def __init__(self, json):
-            self.block=json["result"]["block"]["last_commit"]["precommits"]
-
-    start = block_height - 50
-    end = block_height + 1
-    url_block = []
-
-    for number in range (start, end):
-        block = ("http://"+ i + "/block?height=" + str(number))
-        url_block.append(block)
-
-    total_blocks = len(url_block)
-    counts = dict()
-    print ("Scanning all blocks...")
-    for i in url_block:
-        url_data = json.load(urllib2.urlopen(i))
-        foo = Block (url_data)
-        # print("Got " + i)
-        for k in foo.block:
+            self.blockheight=json["result"]["block"]
+
+    class ValidatorsHeight():
+        def __init__(self, json):
+            self.validatorsheight=json["result"]
+
+
+    def get_height_urls():
+        global url_block, url_validators 
```

Comment:
You need a class call `Blocks` and a class called 'Validators`

they should have attribute called `url_suffix` that is what you add to the of the node url

You should pass into into `__init__ `("the constructor") last_run and the ipaddress of the node.

there should be a method `fetch_data` on each class that downloads the needed data from the node and the stores in  `self.blocks` and `self.validators`

`get_blockheight_valdiators` should then be passed a object of class `Blocks` and an object of class `Valdiators` in it's arguements